Source/WebInspectorUI/UserInterface/Views/LogContentView.js

```diff
@@ -1333,6 +1323,25 @@ WI.LogContentView = class LogContentView extends WI.ContentView
         if (this.element.firstChild !== this._hiddenMessagesBannerElement)
             this.element.insertAdjacentElement("afterbegin", this._hiddenMessagesBannerElement);
     }
+
+    _showConditionallyVisibleScopeBarItemsIfNeeded(newMessage) {
+        for (let scope of WI.LogContentView.ConditionallyVisibleScopes) {
+            if (this._scopeFromMessageLevel(newMessage.level) === scope)
+                this._scopeBar.item(scope).hidden = false;
+        }
+    }
+
+    _hideConditionallyVisibleScopeBarItemsIfNeeded() {
+        for (let scope of WI.LogContentView.ConditionallyVisibleScopes) {
```

Comment:
it's probably better to iterate on `this._allMessageElements()` since that's a `querySelector`, which is more expensive and should ideally not be called more than once
```js
let messageScopes = new Set;

for (let messageElement of this._allMessageElements()) {
    let message = element.__message;
    if (!message)
        continue;

    let scope = this._scopeFromMessageLevel(message.level);
    messageScopes.add(scope);
}

for (let scope of WI.LogContentView.ConditionallyVisibleScopes) {
    let item = this._scopeBar.item(scope);
    if (item.selected)
        continue;

    if (!messageScopes.has(scope))
        item.hidden = true;
}
```

ideally we wouldn't even need to use `this._allMessageElements()`, but that would likely require an extra member variable for "what scopes have we seen so far" and some logic in `_messageAdded` (as well as all the places this is currently called) to keep it up to date